crypto-keys/src/main/java/com/palantir/crypto2/keys/DefaultAsyncKeyStorageStrategy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.crypto2.keys;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public final class DefaultAsyncKeyStorageStrategy implements AsyncKeyStorageStrategy {
+
+    private static final ExecutorService DEFAULT_EXECUTOR = Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder().setNameFormat("async-key-storage-strategy-%s").build());
+
+    private final KeyStorageStrategy keys;
+    private final ExecutorService executor;
+
+    public DefaultAsyncKeyStorageStrategy(KeyStorageStrategy keys) {
+        this(keys, DEFAULT_EXECUTOR);
+    }
+
+    @VisibleForTesting
```

Comment:
think I would make this the default: make it explicit that you need to provide an executor service